src/mpi/datatype/typerep/src/typerep_dataloop_commit.c

```diff
@@ -137,20 +136,22 @@ void MPIR_Typerep_commit(MPI_Datatype type)
     switch (combiner) {
         case MPI_COMBINER_CONTIGUOUS:
         case MPI_COMBINER_VECTOR:
-        case MPI_COMBINER_HVECTOR_INTEGER:
         case MPI_COMBINER_HVECTOR:
         case MPI_COMBINER_INDEXED_BLOCK:
         case MPI_COMBINER_HINDEXED_BLOCK:
         case MPI_COMBINER_INDEXED:
-        case MPI_COMBINER_HINDEXED_INTEGER:
         case MPI_COMBINER_HINDEXED:
-        case MPI_COMBINER_STRUCT_INTEGER:
         case MPI_COMBINER_STRUCT:
             if (ints[0] == 0) {
                 MPIR_Dataloop_create_contiguous(0, MPI_INT, (void **) dlp_p);
                 goto clean_exit;
             }
             break;
+        case MPI_COMBINER_HVECTOR_INTEGER:
```

Comment:
Just thinking about the ABI implications of this change. Does an application always have to query for the combiner type? Could it hardcoded somehow?

Reply:
User do query the combiner with `MPI_Type_get_envelope`.

Reply:
Yeah, I suppose this is truly runtime information. Properly written programs should not be affected.